### PR TITLE
[IOTDB-2705] Separate CQ persistence from MLog

### PR DESCRIPTION
**Pull request by MarcosZyk**
###  Problem
The storage of CQ task has nothing to do with mlog nor even metadata module and it affects the following work of SchemaEngine refactor.

### Solution
Use CQLog to persist CQ operations.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/engine/cq/CQLogWriter.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  private static final String TOO_LARGE_RECORD_EXCEPTION =
+      "Current CQ management operation plan is too large to write into buffer, please increase cqlog_buffer_size.";
+
+  private final ByteBuffer logBuffer;
+  private final LogWriter logWriter;
+
+  public CQLogWriter(String logFilePath) throws IOException {
+    logBuffer = ByteBuffer.allocate(IoTDBDescriptor.getInstance().getConfig().getTlogBufferSize());
+    File logFile = SystemFileFactory.INSTANCE.getFile(logFilePath);
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/engine/cq/CQLogWriter.java:42]: logFile is not closed in this class.

File: server/src/main/java/org/apache/iotdb/db/engine/cq/CQLogWriter.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    } catch (BufferOverflowException e) {
+      throw new IOException(TOO_LARGE_RECORD_EXCEPTION, e);
+    } finally {
+      logBuffer.clear();
+    }
+  }
+
+  @Override
+  public void close() throws IOException {
+    logWriter.close();
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/engine/cq/CQLogWriter.java:72]: logFile should be closed too.

File: server/src/main/java/org/apache/iotdb/db/engine/cq/ContinuousQueryService.java [hunk 3]
```diff
@@ -113,15 +162,26 @@ private void checkAndSubmitTasks() {
...
       }
+
+      continuousQueryPlans.clear();
+
+      if (logWriter != null) {
+        logWriter.close();
+        logWriter = null;
+      }
+    } catch (IOException ignored) {
+
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/engine/cq/ContinuousQueryService.java:184]: We'd better log the exception.

File: server/src/main/java/org/apache/iotdb/db/engine/cq/ContinuousQueryService.java [hunk 4]
```diff
@@ -51,35 +55,80 @@
       ContinuousQueryTaskPoolManager.getInstance();
   private static final long TASK_SUBMIT_CHECK_INTERVAL =
       IoTDBDescriptor.getInstance().getConfig().getContinuousQueryMinimumEveryInterval() / 2;
+
+  private static final String LOG_FILE_DIR =
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/engine/cq/ContinuousQueryService.java:59]: It seems that the dir data/system/cq/cqlog.bin is better.

File: server/src/main/java/org/apache/iotdb/db/engine/cq/ContinuousQueryService.java [hunk 5]
```diff
@@ -51,35 +55,80 @@
...
       new ConcurrentHashMap<>();
   private final ConcurrentHashMap<String, Long> nextExecutionTimestamps = new ConcurrentHashMap<>();
 
+  private CQLogWriter logWriter;
+
+  public void doRecovery() throws StartupException {
+    try {
+      File logFile = SystemFileFactory.INSTANCE.getFile(LOG_FILE_NAME);
+      if (!logFile.exists()) {
+        return;
```
Comment (SteveYurongSu) [inline server/src/main/java/org/apache/iotdb/db/engine/cq/ContinuousQueryService.java:75]: Should we close the logFile here?

File: server/src/main/java/org/apache/iotdb/db/engine/cq/ContinuousQueryService.java (hunk 4 above)
Comment (MarcosZyk) [reply server/src/main/java/org/apache/iotdb/db/engine/cq/ContinuousQueryService.java:59]: Fixed.

File: server/src/main/java/org/apache/iotdb/db/engine/cq/CQLogWriter.java (hunk 1 above)
Comment (MarcosZyk) [reply server/src/main/java/org/apache/iotdb/db/engine/cq/CQLogWriter.java:42]: File object doesn't take any file handler and has no close() method, thus there's no need to close it. It is enough that we make sure CQLogWriter and CQLogReader closed after usage.

**Review by SteveYurongSu — APPROVED** [review]
LGTM

**Pull request merged**